nixos/modules/hardware/steam-hardware.nix

```diff
@@ -21,5 +21,12 @@ in
     services.udev.packages = [
       pkgs.steamPackages.steam
     ];
+
+    # The uinput module needs to be loaded in order to trigger the udev rules
+    # defined in the steam package for setting permissions on /dev/uinput.
+    #
+    # If the udev rules are not triggered, some controllers won't work with
+    # steam.
+    boot.kernelModules = [ "uinput" ];
```

Comment:
Adding this made this steam error go away:
```
Couldnt initialize virtual gamepad: Couldnt open /dev/uinput for writing
```
This is now my `/dev/uinput` permissions:
```
$ ls -al /dev/uinput 
crw-rw----+ 1 root root 10, 223 Oct  6 12:07 /dev/uinput
```